src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs

```diff
@@ -0,0 +1,322 @@
+					_docUri = "file://" + Android.OS.Environment.GetExternalStoragePublicDirectory(Android.OS.Environment.DirectoryMovies).CanonicalPath;
+					break;
+				case PickerLocationId.Unspecified:
+					_docUri = "";
+					break;
+				default:
+					throw new NotImplementedException("FileOpenPicker unimplemented type of initial dir");
+
+			}
+
+		}
+
+		public FileOpenPicker()
+		{
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.Kitkat)
+			{
+				throw new NotImplementedException("FileOpenPicker requires Android KitKat (API level 19) or newer");
+			}
+			SetDocUri();
+			// check permission: READ_EXTERNAL_STORAGE
+		}
+
+		public Foundation.IAsyncOperation<Windows.Storage.StorageFile> PickSingleFileAsync()
+			=> PickSingleFileAsyncTask().AsAsyncOperation<Windows.Storage.StorageFile>();
+
+		public Foundation.IAsyncOperation<IReadOnlyList<Windows.Storage.StorageFile>> PickMultipleFilesAsync()
+			=> PickMultipleFilesAsyncTask().AsAsyncOperation();
+
+		public async Task<Windows.Storage.StorageFile> PickSingleFileAsyncTask()
+		{
+			var pickedFiles = await PickFilesAsync(false);
+			if (pickedFiles is null)
+			{
+				return null;
+			}
+
+			return pickedFiles.ElementAt(0);
+		}
+		public async Task<IReadOnlyList<Windows.Storage.StorageFile>> PickMultipleFilesAsyncTask()
+		{
+			var pickedFiles = await PickFilesAsync(true);
+			if (pickedFiles is null)
+			{
+				return null;
+			}
+
+			return pickedFiles.AsReadOnly();
+		}
+
+		private TaskCompletionSource<List<Windows.Storage.StorageFile>> completionSource;
+
+		private string PathFromUri(Android.Net.Uri fileUri)
+		{
+			// basic file
+			if (fileUri.Scheme.Equals("file", StringComparison.OrdinalIgnoreCase))
+			{
+				return fileUri.Path;
+			}
+
+			// more complicated Uris ("content://")
+			if (Android.Provider.DocumentsContract.IsDocumentUri(Android.App.Application.Context, fileUri))
+			{
+				if (fileUri.Authority.Equals("com.microsoft.skydrive.content.StorageAccessProvider"))
+				{
+					throw new NotImplementedException("FileOpenPicker - Onedrive documents are not implemented yet");
+				}
+
+				string columnName = Android.Provider.MediaStore.Files.FileColumns.Data;
+				string[] projection = { columnName };
+				var cursor = Android.App.Application.Context.ContentResolver.Query(fileUri, projection, null, null, null);
+				if (cursor is null)
+					throw new Exception("FileOpenPicker - cannot get cursor");
+
+				if (!cursor.MoveToFirst())
+					throw new Exception("FileOpenPicker - cannot MoveToFirst");
+
+				int columnNo = cursor.GetColumnIndex(columnName);
+				if (columnNo < 0)
+					throw new Exception("FileOpenPicker - column with data doesn't exist?");
+
+				string filePath = cursor.GetString(columnNo);
+				cursor.Close();
+				cursor.Dispose();
+
+				if (!string.IsNullOrEmpty(filePath))
+				{
+					return filePath;
+				}
+
+			}
+			throw new NotImplementedException("FileOpenPicker - not implemented document type");

```

Comment:
```suggestion
			throw new NotSupportedException("FileOpenPicker - not implemented document type");

```

Indicated in the error string what format is not supported.